homeassistant/components/sensor/luftdaten.py

```diff
@@ -95,48 +104,53 @@ def name(self):
     @property
     def state(self):
         """Return the state of the device."""
-        return self._state
+        try:
+            return self.luftdaten.data.values[self.sensor_type]
+        except AttributeError:
+            pass
 
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
         return self._unit_of_measurement
 
-    def update(self):
-        """Get the latest data from REST API and update the state."""
-        self.rest_client.update()
-        value = self.rest_client.data
-
-        if value is None:
-            self._state = None
-        else:
-            parsed_json = json.loads(value)
-
-            log_entries_count = len(parsed_json) - 1
-            latest_log_entry = parsed_json[log_entries_count]
-            sensordata_values = latest_log_entry['sensordatavalues']
-            for sensordata_value in sensordata_values:
-                if sensordata_value['value_type'] == self.sensor_type:
-                    self._state = sensordata_value['value']
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if self.luftdaten.data.meta is None:
+            return
+
+        attr = {
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+            ATTR_SENSOR_ID: self._sensor_id,
+            'lat': self.luftdaten.data.meta['latitude'],
+            'long': self.luftdaten.data.meta['longitude'],
+        }
+        return attr
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Get the latest data from luftdaten.info and update the state."""
+        try:
+            yield from self.luftdaten.async_update()
+        except TypeError:
```

Comment:
When could this error happen?

Reply:
All sensors are under the control of their owner and if one decided to changing something it can happen that several update windows of the Luftdate API are missed. There are sensors which are not properly implemented and only send partial data. Also, there is no way to tell if the sensor is temporary offline or permanent.

This can be removed if the Luftdaten API is declared stable and the module implements more error handling.